Source/WebKit/UIProcess/ProvisionalPageProxy.cpp

```diff
@@ -217,19 +217,21 @@ void ProvisionalPageProxy::initializeWebPage(RefPtr<API::WebsitePolicies>&& webs
         RegistrableDomain navigationDomain(m_request.url());
         if (m_page->openerFrame())
             mainFrameIdentifier = m_page->mainFrame()->frameID();
-        if (auto existingRemotePageProxy = m_browsingContextGroup->takeRemotePageInProcessForProvisionalPage(page(), navigationDomain)) {
-            m_webPageID = existingRemotePageProxy->pageID();
-            m_mainFrame = existingRemotePageProxy->page()->mainFrame();
-            m_messageReceiverRegistration.stopReceivingMessages();
-            m_messageReceiverRegistration.transferMessageReceivingFrom(existingRemotePageProxy->messageReceiverRegistration(), *this);
-            LocalFrameCreationParameters localFrameCreationParameters {
-                std::nullopt
-            };
-            protectedProcess->send(Messages::WebPage::TransitionFrameToLocal(localFrameCreationParameters, m_page->mainFrame()->frameID()), m_webPageID);
-            m_needsCookieAccessAddedInNetworkProcess = true;
-            registerWithInspectorController = false; // FIXME: <rdar://121240770> This is a hack. There seems to be a bug in our interaction with WebPageInspectorController.
+        if (m_browsingContextGroup.ptr() == &m_page->browsingContextGroup()) {
```

Comment:
This check shouldn't be necessary.  If we've swapped browsing context groups, it won't have an existing RemotePageProxy.